packages/myst-cli/src/utils/imagemagick.ts

```diff
@@ -135,7 +138,9 @@ export async function convertImageToWebp(
     return null;
   }
   const imageExt = path.extname(image).toLowerCase();
-  const allowedExtensions = ['.png', '.jpg', '.jpeg', '.tiff', '.gif', '.pdf'];
+  // A race condition may execute this before the PDF --> PNG conversion is complete
```

Comment:
I thought this was handled by performing pdf -> png conversion as part of this function - https://github.com/executablebooks/mystmd/blob/3e2e930490c4ff802673983bac20099b9ecf8757/packages/myst-cli/src/utils/imagemagick.ts#L206

I would expect that line to ensure png conversion is complete before running the webp conversion 🤔

Reply:
Putting that back in would mean we would also need the `-trim` command here: https://github.com/executablebooks/mystmd/blob/3e2e930490c4ff802673983bac20099b9ecf8757/packages/myst-cli/src/utils/imagemagick.ts#L189

Reply:
Oh - I see from the changeset text, the problem is just overwriting with this command - so, maybe if we add `-trim` here, all will be fine. The png will be overwritten, but it will be overwritten with the correct image.

Reply:
I was pretty weirded out how we even get to having the PDF in the webp conversion part, and is hinting at a better way to do our image transformations, I think it could be that this is running in two articles and the conversions are running twice (expected) - and then somehow a third time when a webp-conversion is triggered.

I really don't think that the webp function should handle pdf --> png conversion at all. 🤔

Maybe safe for now to just put the `-trim` in this duplicate function.

Reply:
I have backed this out a bit now!